src/FileSig.hs

```diff
@@ -1,4 +1,4 @@
-{-# LANGUAGE TemplateHaskell #-}
+-- {-# LANGUAGE TemplateHaskell #-}
```

Comment:
There might be some gains to be had with TH or QuasiQuotes, to do computations at compile time (since the map is static) so that the `MagicMap` and `longestSignature :: Int` are just that, instead of computations to get the result.

Probably need some benchmarking to see if it has any impact.

Reply:
Oh I see, you're talking about just sorting the map at compile time so every time it's used we don't sort by longest to shortest signature? That makes sense.

Reply:
Yeah `magicMap` at the moment is a computation that makes a `MagicMap` out of the `rawMagicMap`, but I hope after compilation it is literally just a `HashMap` instead of a computation to make one.
Though, knowing Haskell, even though it might have to construct the magicMap first, it'll only do that once. After that it just reuses the already built map. (or someone uses the really weird setup in which the references get dropped after every use... but that's uncommon... I think)